Lib/test/test_asyncio/test_futures.py

```diff
@@ -283,6 +279,34 @@ def test_exception(self):
         self.assertRaises(asyncio.InvalidStateError, f.set_exception, None)
         self.assertFalse(f.cancel())
 
+    def test_stop_iteration_exception(self):
```

Comment:
Tests are almost duplicates. You can share the code. For example:
```
```suggestion
    def test_stop_iteration_exception(self, StopIteration=StopIteration):
```
And then in `test_stop_iteration_subclass_exception` call `test_stop_iteration_exception(MyStopIteration)`.